### PR TITLE
Fix #1285: FIDO2: Review provider API

### LINKED ISSUE #1285: FIDO2: Review provider API
Review the provider API and clean it up.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthRegistrationProvider.java [hunk 1]
```diff
@@ -75,7 +75,7 @@ public RegistrationChallenge provideChallengeForRegistration(String userId, Stri
 
     @Override
     @Transactional
-    public RegistrationChallenge provideChallengeForRegistrationChallengeValue(String applicationId, String challengeValue) throws Fido2AuthenticationFailedException {
+    public RegistrationChallenge findRegistrationChallengeByValue(String applicationId, String challengeValue) throws Fido2AuthenticationFailedException {
```
Comment (banterCZ) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthRegistrationProvider.java:78]: Out of the scope of this PR: May this method be `@Transacional(readonly = true)`?

**Review by banterCZ — DISMISSED** [review]
Thanks, the naming is improved.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthRegistrationProvider.java (hunk 1 above)
Comment (romanstrobl) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/fido2/PowerAuthRegistrationProvider.java:78]: I will review the `@Transactional` annotations, good point.

**Review by banterCZ — APPROVED** [review]
Thanks for marking transactions read-only.

**Pull request merged**